velox/connectors/hive/tests/HivePartitionUtilTest.cpp

```diff
@@ -117,3 +117,47 @@ TEST_F(HivePartitionUtilTest, partitionName) {
         "Unsupported partition type: MAP");
   }
 }
+
+TEST_F(HivePartitionUtilTest, partitionNameForNull) {
+  std::vector<std::string> partitionColumnNames{
+      "flat_bool_col",
+      "flat_tinyint_col",
+      "flat_smallint_col",
+      "flat_int_col",
+      "flat_bigint_col",
+      "dict_string_col",
+      "const_date_col"};
+
+  RowVectorPtr input = makeRowVector(
+      partitionColumnNames,
+      {makeNullableFlatVector<bool>(
+           std::vector<std::optional<bool>>{std::nullopt}),
+       makeNullableFlatVector<int8_t>(
+           std::vector<std::optional<int8_t>>{std::nullopt}),
+       makeNullableFlatVector<int16_t>(
+           std::vector<std::optional<int16_t>>{std::nullopt}),
+       makeNullableFlatVector<int32_t>(
+           std::vector<std::optional<int32_t>>{std::nullopt}),
+       makeNullableFlatVector<int64_t>(
+           std::vector<std::optional<int64_t>>{std::nullopt}),
+       makeNullableFlatVector<StringView>(
+           std::vector<std::optional<StringView>>{std::nullopt}),
+       makeConstant<int32_t>(std::nullopt, 1, DATE())});
+
+  for (auto i = 1; i <= partitionColumnNames.size(); i++) {
+    std::vector<column_index_t> partitionChannels(i);
+    std::iota(partitionChannels.begin(), partitionChannels.end(), 0);
+    auto partitionEntries = extractPartitionKeyValues(
+        makePartitionsVector(input, partitionChannels), 0);
+    for (const auto& entry : partitionEntries) {
+      EXPECT_EQ(entry.second, "");
+    }
+    auto expectedPartitionNames = folly::join(
```

Comment:
expectedPartitionNames -> expectedPartitionName (no 's' at the end)

Why do you feel the need to test makePartName here? Don't we have a separate unit test for makePartName to verify that it converts empty values to __HIVE_DEFAULT_PARTITION__?

Anyway, this test only verifies the case when all partition key values are null, but doesn't test cases where some are null and others are not.

Reply:
I did not realize we have a separate unit test for makePartName. I agree it's unnecessary, removed.